pkg/linode-client/linode-client.go

```diff
@@ -23,17 +25,49 @@ type LinodeClient interface {
 	DeleteVolume(context.Context, int) error
 
 	ResizeVolume(context.Context, int, int) error
+
+	NewEventPoller(context.Context, any, linodego.EntityType, linodego.EventAction) (*linodego.EventPoller, error)
 }
 
-func NewLinodeClient(token, ua string, url string) *linodego.Client {
+func NewLinodeClient(token, ua string, apiURL string) (*linodego.Client, error) {
 	// Use linodego built-in http client which supports setting root CA cert
 	linodeClient := linodego.NewClient(nil)
 	linodeClient.SetUserAgent(ua)
 	linodeClient.SetToken(token)
 
-	if url != "" {
-		linodeClient.SetBaseURL(url)
+	if apiURL != "" {
+		host, version, err := getAPIURLComponents(apiURL)
+		if err != nil {
+			return nil, err
+		}
+
+		linodeClient.SetBaseURL(host)
+
+		if version != "" {
+			linodeClient.SetAPIVersion(version)
+		}
+	}
+
+	return &linodeClient, nil
+}
+
+// getAPIURLComponents returns the API URL components (base URL, api version) given an input URL.
+// This is necessary due to some recent changes with how linodego handles
+// client.SetBaseURL(...) and client.SetAPIVersion(...)
+func getAPIURLComponents(apiURL string) (string, string, error) {
```

Comment:
Hey @lgarber-akamai, @luthermonson, I don't think this is quite aligned with the way we want to move forward. We'll be bumping linodego across our components and we can't introduce this everywhere, I was under the impression we wanted to fix this at the ENV level, not in code.

(This linodego ticket is currently being worked on.)